### PR TITLE
8256181: Remove Allocation of old generation on alternate memory devices functionality

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I get reviews for this change that removes the "Allocation of old generation of Java heap on alternate memory devices" functionality introduced with JDK 12 with [JDK-8202286](https://bugs.openjdk.java.net/browse/JDK-8202286) due to being

- not used by anyone
- not maintained by anyone, i.e. several bugs open for a long time and bit rotting
- requiring some workarounds for new feature development wrt to heap management

All flags covered by this feature were experimental flags, so there are no additional procedural issues to take.

I tried to remove all but a few minor cleanups that I thought useful, but of course this is very subjective.

Testing: hs-tier1-5

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8256181](https://bugs.openjdk.java.net/browse/JDK-8256181): Remove Allocation of old generation on alternate memory devices functionality

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Author) ⚠️ Review applies to 222739bc6332e684b42a146d3d369e121803547d
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**)
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1162/head:pull/1162`
`$ git checkout pull/1162`

### CODE REVIEW

File: test/hotspot/jtreg/TEST.ROOT [hunk 1]
```diff
@@ -75,7 +75,6 @@ requires.properties= \
     vm.compiler2.enabled \
     vm.musl \
     docker.support \
-    test.vm.gc.nvdimm \
```
Comment (albertnetymk) [inline test/hotspot/jtreg/TEST.ROOT:78]: `test/jtreg-ext/requires/VMProps.java` still references `test.vm.gc.nvdimm`, btw.

**Review by albertnetymk — APPROVED** [review]
A general comment for future PRs: I think it's best to isolate mechanical changes into their own commits; e.g. `HeapRegionManager* _hrm;` -> `HeapRegionManager _hrm;`. Otherwise, a real change, buried in the immense size of diff, might slip through.

File: test/hotspot/jtreg/TEST.ROOT (hunk 1 above)
Comment (tschatzl) [reply test/hotspot/jtreg/TEST.ROOT:78]: The original change introduced the `_hrm` -> `*_hrm` change, so this is a straightforward reversal, no optimization :)

There has been one more removal possible by the `test.vm.gc.nvdimm` line in VMProps.java.

Thanks for your review!

**Comment by iklam** [comment]
This is a good fix. It removes the dependency on the problematic API `ReservedSpace::first_part(..., split=true)`, which cannot be implemented correctly on Windows (see [JDK-8256079](https://bugs.openjdk.java.net/browse/JDK-8256079)). This API is used only by CDS (to be removed in [JDK-8255917](https://bugs.openjdk.java.net/browse/JDK-8255917)) and heterogenous heap. So hopefully we can remove this problematic API soon.

**Review by iignatev — APPROVED** [review]

**Review by iklam — APPROVED** [review]
LGTM

**Comment by tschatzl** [comment]
Thanks @albertnetymk @iklam @iignatev for your reviews.

/integrate

**Pull request closed without merging**